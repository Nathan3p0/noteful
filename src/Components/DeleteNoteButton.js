import React from 'react';
import { NotefulContext } from './NotefulContext'

const DeleteNoteButton = (props) => {
    return ( 
        <NotefulContext.Consumer>
            {({deleteNote}) => (
            <button onClick={() => deleteNote(props.noteId)}>Delete Note</button>
            )}
        </NotefulContext.Consumer>
     );
}
 
export default DeleteNoteButton;