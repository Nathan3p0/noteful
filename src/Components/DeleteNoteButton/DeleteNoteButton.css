.btnDelete {
    border: 1px solid #939393;
    padding: 5px;
    background-color: white;
    color: #cb132d;
}

.btnDelete:hover {
    border: 0px solid black;
    padding: 5px;
    background-color: #cb132d;
    color: white;
}