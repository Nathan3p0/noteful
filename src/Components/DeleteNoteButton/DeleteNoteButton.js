import React from 'react';
import { NotefulContext } from '../NotefulContext';
import './DeleteNoteButton.css'

const DeleteNoteButton = (props) => {
    return ( 
        <NotefulContext.Consumer>
            {({deleteNote}) => (
            <button className="btn btnDelete" onClick={() => deleteNote(props.noteId)}>Delete Note</button>
            )}
        </NotefulContext.Consumer>
     );
}

DeleteNoteButton.defaultProps = {
    noteId : ''
}
 
export default DeleteNoteButton;