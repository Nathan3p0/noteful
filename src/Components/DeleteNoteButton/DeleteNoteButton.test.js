import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import DeleteNoteButton from './DeleteNoteButton'


describe('Delete Note Button Component', () => {

    const buttonProps = {
        noteId: "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
    }

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DeleteNoteButton />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders empty', () => {
        const tree = renderer.create(<DeleteNoteButton />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders properly when given props', () => {
        const tree = renderer.create(<DeleteNoteButton noteId={buttonProps.noteId} />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})