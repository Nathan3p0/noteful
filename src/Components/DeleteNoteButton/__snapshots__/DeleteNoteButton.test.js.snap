// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Note Button Component renders empty 1`] = `
<button
  className="btn btnDelete"
  onClick={[Function]}
>
  Delete Note
</button>
`;

exports[`Delete Note Button Component renders properly when given props 1`] = `
<button
  className="btn btnDelete"
  onClick={[Function]}
>
  Delete Note
</button>
`;
